File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def running_sum(nums)
    array = []
    sum = 0
    index = 0
    while index < nums.length do
        sum = sum + nums[index]
        array << sum
        index += 1
    end
    return array
end

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.py
class Solution(object):
    def runningSum(self, nums):
        new_array = []
        running_total = 0
        index = 0
        while index < len(nums):
            running_total += nums[index]
            new_array.append(running_total)
            index += 1
        return new_array
            
        

# loop though array
# add running total plus array at index + 1
# push running total to new array